File: gaia-sdk-javascript/src/graphql/GaiaRequest.ts
import {Query} from "./request/type/Query";
import {Mutation} from "./request/type/Mutation";
import {Subscription} from "./request/type/Subscription";

export class GaiaRequest {
    public static query(config:(_:Query) => void) {
        const request = new Query();
        config(request);
        return request;
    }
    public static mutation(config:(_:Mutation) => void) {
        const request = new Mutation();
        config(request);
        return request;
    }
    public static subscription(config:(_:Subscription) => void) {
        const request = new Subscription();
        config(request);
        return request;
    }
}

File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/GaiaRequest.kt
package gaia.sdk

import gaia.sdk.request.type.Query
import gaia.sdk.request.type.Mutation
import gaia.sdk.request.type.Subscription

class GaiaRequest {

    companion object {
        @JvmStatic
        fun query(config: Query.() -> Unit) = Query().apply(config)

        @JvmStatic
        fun mutation(config: Mutation.() -> Unit) = Mutation().apply(config)

        @JvmStatic
        fun subscription(config: Subscription.() -> Unit) = Subscription().apply(config)
    }

}
